src/a11y-dialog.ts

```diff
@@ -221,11 +221,12 @@ function $$(selector: string, context: ParentNode = document): HTMLElement[] {
  * Trap the focus inside the given element
  */
 function trapTabKey(el: HTMLElement, event: KeyboardEvent) {
-  const focusableChildren = getFocusableChildren(el)
-  const firstFocusableChild = focusableChildren[0]
-  const lastFocusableChild = focusableChildren.at(-1)
+  const [firstFocusableChild, lastFocusableChild] =
+    getFirstAndLastFocusableChild(el)
 
-  const activeElement = getActiveElement()
+  if (!firstFocusableChild || !lastFocusableChild) return
+
+  const activeElement = getActiveElement() || document.activeElement
```

Comment:
🔍 This fallback is necessary for elements like `<fancy-div tabindex="0">`, where the element will have a shadowRoot, but nothing in there is focusable, so we have to fall back to the thing that is focusable in light DOM – the host itself.

Reply:
Please kindly add that comment *in* the code. ;)